kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/CommandGatewayExtensions.kt

```diff
@@ -19,43 +19,28 @@
  */
 package org.axonframework.extensions.kotlin
 
-import org.axonframework.commandhandling.CommandCallback
 import org.axonframework.commandhandling.CommandMessage
 import org.axonframework.commandhandling.gateway.CommandGateway
 import org.axonframework.messaging.MetaData
 import java.util.concurrent.TimeUnit
 
```

Comment:
Can you explain me why there isn't an extension function for the `CommandGateway.send(Object)` method which returns a `CompletableFuture`. Is that not necessary somehow?

Reply:
In general, it is not the idea to replace all the already existing methods with Kotlin based extension methods but provide more Kotlin idiomatic methods in cases where it makes sense. In this case, there is nothing to be gained by making an extension method to proxy to `send(Object)`.

Having said that, I did more testing and investigation of `sendAndWaitWithResponse` extension methods and found that at the moment they just simply proxy the call. Reified generics, in this case, are only used as regular generics and these methods can be used the same way regular `sendAndWait` are. There are no visible benefits in either type inference or invocation since a generic parameter is required to invoke both `sendAndWait` and `sendAndWaitWithResponse` if used as a statement and not an expression.

I would be in favour of removing current `sendAndWaitWithResponse` extensions unless there is a reason to keep them, such as a use-case that I missed.